packages/@aws-cdk/toolkit/test/actions/deploy.test.ts

```diff
@@ -0,0 +1,122 @@
+import * as s3 from 'aws-cdk-lib/aws-s3';
+import * as core from 'aws-cdk-lib/core';
+import { Toolkit } from '../../lib/toolkit';
+import { fixture, TestIoHost } from '../_helpers';
+import { CloudFormationClient, DescribeChangeSetCommand, DescribeStacksCommand, StackStatus } from '@aws-sdk/client-cloudformation';
+import { mockClient } from 'aws-sdk-client-mock';
+import { StackSelectionStrategy } from '../../lib';
+
+const ioHost = new TestIoHost();
+const notifySpy = jest.spyOn(ioHost, 'notify');
+const requestResponseSpy = jest.spyOn(ioHost, 'requestResponse');
+const cdk = new Toolkit({ ioHost });
+const mockCloudFormationClient = mockClient(CloudFormationClient);
+
+const cxFromBuilder = async () => {
+  return cdk.fromAssemblyBuilder(async () => {
+    const app = new core.App();
+    new core.Stack(app, 'Stack1');
+    new core.Stack(app, 'Stack2');
+
+    // @todo fix api
+    return app.synth() as any;
+  });
+};
+
+const cxFromApp = async (name: string) => {
+  return cdk.fromCdkApp(`node ${fixture(name)}`);
+};
+
+beforeEach(() => {
+  requestResponseSpy.mockClear();
+  notifySpy.mockClear();
+  mockCloudFormationClient.reset();
```

Comment:
all this mockCloudFormation stuff is copied over. there's a whole bunch more to mock as we add more unit tests...